common/unsafe/src/main/java/org/apache/spark/unsafe/memory/UnsafeMemoryAllocator.java

```diff
@@ -36,22 +42,35 @@ public MemoryBlock allocate(long size) throws OutOfMemoryError {
 
   @Override
   public void free(MemoryBlock memory) {
-    assert (memory.obj == null) :
+    if (memory == OffHeapMemoryBlock.NULL) return;
+    assert(memory instanceof OffHeapMemoryBlock);
```

Comment:
I think this assert should be in front of `if (memory == OffHeapMemoryBlock.NULL) return;`?

Reply:
good catch